### DISCUSSION TITLE
Is it possible to get isValidating flag updated upon first submit?

### DISCUSSION THREAD

**Discussion by vicke4** [react-hook-form/react-hook-form#5369] [Q&A] [answered]
Hi,

I use `useForm` with the default settings, meaning `mode` is set to `onSubmit`. Hence, the form validation is triggered only after submitting the form for the first time.

I'm doing an async validation on one of the fields, and it takes some seconds to complete. I want to let the user know that the validation is in progress. But, on the first submit the `formState` is updated only after the validation completes.

I don't want to pass `onChange` to `mode`. I have tried calling `trigger` just before submitting. But, in that case, the form is in `isValidating` state event after the completion of validation. I'm not sure if it's a bug.

Is there anything I can do differently?

**Comment by bluebill1049** [comment, marked as answer]
hey @vicke4 can you share a codesandbox? i can do some investigate and see if this is a bug or not.

**Reply by vicke4** [reply]
At first, I thought the issue is due to the debouncing function I am using. It doesn't work for the first submit alone. But, after the first submit, the `isValidating` flag is updating successfully.

Steps to reproduce:
1. Fill anything in the name field and submit. You will see the `isValidating` flag remains true forever.
2. If you update the value now, the `isValidating` updates as expected.

[CodeSandbox Link](https://codesandbox.io/s/frosty-waterfall-yuueh?file=/src/app.tsx)

**Reply by bluebill1049** [reply]
```
const manualSubmit = () => {
    trigger();
    handleSubmit(onSubmit)();
  };
  ```
  
  curious to know why calling trigger and handleSubmit together?

**Reply by vicke4** [reply]
Thanks for the help.

When I submit the form for the first time, the `isValidating` flag is not updating so I call `trigger` as a workaround.

Should `isValidating` flag update upon first submit?

**Reply by bluebill1049** [reply]
should use `isSubmitting`?

**Reply by vicke4** [reply]
Oh my God, Bill. How did I fail to notice the `isSubmitting` flag 🤦‍♂️. Thanks much!
